agent/util/expansions.go

```diff
@@ -14,6 +14,10 @@ import (
 type DynamicExpansions struct {
 	util.Expansions
 	mu sync.RWMutex
+
+	// redact stores expansions that should be redacted.
+	redact   []string
+	redactMu sync.RWMutex
```

Comment:
I added a separate mutex since it's used separately than the expansions mostly

Reply:
Might be a small edge case but it may be more correct to lock the original mutex and make `PutAndRedact` hold a single lock for the entire duration because putting expansions and redacting should logically be one atomic operation (holding a write lock the entire time). If you let go of the lock in between put and redact, or use separate locks for expansions vs. redaction, the operation won't be atomic, so one goroutine could see that an expansion has been added but wouldn't see that it's redacted.

Realistically I can't see this happening right now, but may prevent future bugs if we change how commands run.

Reply:
That makes a lot of sense! I'll switch to a single one.

paradoxically I was thinking that two mutexs allows it to be written asap and read despite the redaction not present.

I was thinking alternatively, could edit so that when putting and redacting, do some double locking before redacting but seems like too much work-

Reply:
I would say we generally don't need to worry about performance implications of in-memory mutexes unless we know there's significant contention because the performance difference of using them for quick read/write accesses on an in-memory map is pretty much nonexistent compared to the other more expensive things that Evergreen does.